### ISSUE TITLE
Numeric type is broken for small negative values

### ISSUE THREAD

**Issue by kirk-anchor** [jackc/pgtype#210]
`pgtype.Numeric` does not work correctly for small negative values, e.g. `-0.000123`
```golang
	n := pgtype.Numeric{}
	n.Set("-0.000123")
	s := ""
	n.AssignTo(&s)
	fmt.Println(s)
```
Results in: `0.00-123` instead of `-0.000123`

**Comment by jschaf** [comment]
@jackc This looks like a pretty bad bug. I think pgtype.Numeric to float64 conversion is wrong for all values with an exponent of 1 divisible by 10, meaning `[10, 20, 30, ..., 90]`

**Comment by jackc** [comment]
Thanks @jschaf. Fix is merged and new release `v1.14.2` is tagged.
